Use actions/setup-node to cache pnpm store instead of doing it manually

dtinth opened the pull request:
`actions/setup-node` supports the option `cache: pnpm`, so we can take advantage of that to make our workflow simpler.

<img width="878" alt="image" src="https://user-images.githubusercontent.com/193136/224923775-68a7974c-f4b3-48bd-b362-32279239f41b.png">

**Note:** `pnpm/action-setup` must be run BEFORE `actions/setup-node` because otherwise the `setup-node` action will fail due to not being able to find the pnpm executable.

robotkutya commented:
Anyone knows what the difference is between the two methods?
https://github.com/pnpm/action-setup/issues/82

nicolassanmar commented:
I don't know if I have set this up incorrectly, but I find the manual way to be faster than using the cache from `setup-node`.

The manual way does not require us to install the packages again, while this suggested one does:

### Manual (~20s)
![image](https://github.com/pnpm/action-setup/assets/57573579/b9963604-8e2f-4654-9f5f-9136f14de585)


### Using `setup-node` (~1m)

(This was a cache hit)
![image](https://github.com/pnpm/action-setup/assets/57573579/36b5d4b5-3fbf-4545-b563-ab01de9f6541)

dtinth commented:
@nicolassanmar Does your “manual” setup also cache `node_modules`? Using `cache: pnpm` merely caches the pnpm store, not `node_modules`.

This is how I interpret the log message:

| Log message | Description |
| --- | --- |
| `resolved X, reused 0, downloaded X, added X, done` | Cache is not working. |
| `resolved X, reused X, downloaded 0, added X, done` | Store cache is working. |
| `Already up to date` | `node_modules` is cached. |

nicolassanmar commented:
@dtinth You are 100% right. I thought that https://github.com/pnpm/action-setup#use-cache-to-reduce-installation-time suggested caching `node_modules` but I was mistaken.
I am manually caching `node_modules` and I don't think `setup-node` provides a way to do that

robotkutya commented:
thanks guys for the discussion, makes it much more clear!

I think [more info about this would be appreciated](https://github.com/pnpm/action-setup/issues/82) to newcomers, any ideas how to improve the docs in this regard?

FFdhorkin commented:
I would highly suggest you show both as two different options - this solution won't work if you're using a self-hosted runner that has node pre-installed

davidlj95 commented:
`pnpm` docs also mention using `setup-node` to cache `pnpm` store

https://pnpm.io/continuous-integration#github-actions

I think we should update the example to go the recommended way by using `setup-node`. I think that would be the recommended way given you don't have to maintain a custom cache. `setup-node` takes care of that for you (what if `pnpm store` command is deprecated and now another one is used? `setup-node` would take care of that). So a thing less to worry about

Links:
- [Docs source](https://github.com/pnpm/pnpm.io/blob/43457080f72fbf3a123f4297e23a1369715076e6/docs/continuous-integration.md#github-actions)
- [Docs source PR where GitHub Actions + `setup-node` is added](https://github.com/pnpm/pnpm.io/pull/170)

WolfSoko commented:
> > FYI setup-node action has removed `cache: pnpm` in the latest version. You need to specify the cache key manually.
> 
> Hmm I see [it's still there](https://github.com/actions/setup-node/blob/v6.0.0/action.yml#L24-L25) 🤔 [I'm using it like that with no cache key](https://github.com/davidlj95/ngx/blob/c60cd12fa120d5203a06b4742c3888bef0e7b094/.github/actions/setup/action.yml#L8-L12). [And it works](https://github.com/davidlj95/ngx/actions/runs/20656136088/job/59309303741)

Okay, my fault. Maybe I missinterpreted this: https://github.com/actions/setup-node?tab=readme-ov-file#breaking-changes-in-v6

So `cache: true` will NOT automatically work for pnpm (did it work before?), but `cache: pnpm` is still working. No need for manually declaring the cache key and folder.
